sources/lib/structures/ty.mli

```diff
@@ -76,15 +79,25 @@ and trecord = {
   mutable lbs :  (Hstring.t * t) list;
   (** List of fields of the record. Each field has a name,
       and an associated type. *)
+  constr : Hstring.t; (* for ADTs that become records. default is "{" *)
 }
 (** Record types. *)
 
+type adt_constr =
+  (*  { constr : Hstring.t ; destrs : (Hstring.t * t) list }*)
+  Hstring.t * (Hstring.t * t) list
+
+
+type type_body =
```

Comment:
What is the use case envisionned for this type ? A quick search seems to say that this type is not used anywhere except the `type_body` function, particularly, it is not used outside of the `Ty` module.